src/Microsoft.ML.Api/ApiUtils.cs

```diff
@@ -28,18 +28,27 @@ private static OpCode GetAssignmentOpCode(Type t)
             }
 
             // Simple primitive types.
-            if (t == typeof(Single))
-                return OpCodes.Stind_R4;
-            if (t == typeof(Double))
-                return OpCodes.Stind_R8;
-            if (t == typeof(sbyte) || t == typeof(byte) || t == typeof(bool))
-                return OpCodes.Stind_I1;
-            if (t == typeof(short) || t == typeof(ushort))
-                return OpCodes.Stind_I2;
-            if (t == typeof(int) || t == typeof(uint))
-                return OpCodes.Stind_I4;
-            if (t == typeof(long) || t == typeof(ulong))
-                return OpCodes.Stind_I8;
+            switch (t)
+            {
+                case t == typeof(Single):
+                    return OpCodes.Stind_R4;
```

Comment:
Hi @Jungmaus, thanks for the PR! There are a couple things you've probably noted.

You'll see the build failed, for various reasons, including the fact that this code written here is not valid C# of any current version. It *almost* resembles the changes to switch-case statements that happened in C# 7.0, but needs some work, specifically:

```csharp
switch (t)
{
    case Type tt when tt == typeof(Single):
        return OpCodes.Stind_R4;
    case Type tt when tt == typeof(Double):
        return OpCodes.Stind_R8;
    case Type tt when tt == typeof(sbyte) || tt == typeof(byte) || tt == typeof(bool):
        return OpCodes.Stind_I1;
    case Type tt when tt == typeof(short) || tt == typeof(ushort):
        return OpCodes.Stind_I2;
    case Type tt when tt == typeof(int) || tt == typeof(uint):
        return OpCodes.Stind_I4;
    case Type tt when tt == typeof(long) || tt == typeof(ulong):
        return OpCodes.Stind_I8;
}
```

(I also took the liberty of stripping out the pointless `break`s as we see.) However, we should not welcome this. The above code is considerably more verbose and harder to read than the `if` statement it would hypothetically replace.